src/wallet/rpcwallet.cpp

```diff
@@ -3324,6 +3476,124 @@ UniValue z_getbalance(const UniValue& params, bool fHelp)
     return ValueFromAmount(nBalance);
 }
 
+UniValue z_getbalanceforaddress(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() < 1 || params.size() > 2)
+        throw runtime_error(
+            "z_getbalanceforaddress \"address\" ( minconf )\n"
+            "\nReturns the per-protocol (pool) balances of a Unified Address belonging to the node's wallet."
+            "\nArguments:\n"
+            "1. \"address\"      (string) The selected address. It may be a transparent, shielded, or Unified address.\n"
+            "2. minconf        (numeric, optional, default=1) Only include transactions confirmed at least this many times.\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"pools\": {\n"
+            "    \"transparent\": {\n"
+            "        \"valueZat\": amount   (numeric) The amount held in the transparent protocol by this account\n"
+            "    \"},\n"
+            "    \"sprout\": {\n"
+            "        \"valueZat\": amount   (numeric) The amount held in the sprout protocol by this account\n"
+            "    \"},\n"
+            "    \"sapling\": {\n"
+            "        \"valueZat\": amount   (numeric) The amount held in the sapling protocol by this account\n"
+            "    \"},\n"
+            "    \"orchard\": {\n"
+            "        \"valueZat\": amount   (numeric) The amount held in the orchard protocol by this account\n"
+            "    \"}\n"
+            "  \"}\n"
+            "}\n"
+            "Result amounts are in units of " + MINOR_CURRENCY_UNIT + ".\n"
+            "Pools for which the balance is zero are not shown.\n"
+            "\nExamples:\n"
+            "\nThe per-protocol amount received by address \"myaddress\" with at least 1 block confirmed\n"
+            + HelpExampleCli("z_getbalanceforaddress", "\"myaddress\"") +
+            "\nThe per-protocol amount received by address \"myaddress\" with at least 5 blocks confirmed\n"
+            + HelpExampleCli("z_getbalanceforaddress", "\"myaddress\" 5") +
+            "\nAs a json rpc call\n"
+            + HelpExampleRpc("z_getbalanceforaddress", "\"myaddress\", 5")
+        );
+
+    if (!fExperimentalOrchardWallet) {
+        throw JSONRPCError(RPC_WALLET_ENCRYPTION_FAILED, "Error: the Orchard wallet experimental extensions are disabled.");
+    }
+    UniValue pools(UniValue::VOBJ);
+    pools.pushKV("transparent", 99999.99);
+    pools.pushKV("sprout", 99999.99);
+    pools.pushKV("sapling", 99999.99);
+    pools.pushKV("orchard", 99999.99);
+
+    UniValue result(UniValue::VOBJ);
+    result.pushKV("protocolBalances", pools);
```

Comment:
```suggestion
    result.pushKV("poolBalances", pools);
```

Reply:
I think `pool_balances`, as you suggested above (line 3549) would be better. Both conventions already exist (CamelCase and snake_case).